spark-3.5/clickhouse-spark/src/main/scala/org/apache/spark/sql/clickhouse/SchemaUtils.scala

```diff
@@ -110,7 +110,7 @@ object SchemaUtils {
       }
 
   private[clickhouse] def maybeNullable(chType: String, nullable: Boolean): String =
-    if (nullable) wrapNullable(chType) else chType
+    if (nullable && !chType.startsWith("Array") && !chType.startsWith("Map")) wrapNullable(chType) else chType
```

Comment:
If possible, prefer using strong data type rather than string for type matching, for this case, better to handle it on the caller's side

Reply:
ok